src/main/java/com/iexec/core/task/TaskUpdateManager.java

```diff
@@ -0,0 +1,612 @@
+import com.iexec.core.chain.IexecHubService;
+import com.iexec.core.chain.adapter.BlockchainAdapterService;
+import com.iexec.core.replicate.Replicate;
+import com.iexec.core.replicate.ReplicatesList;
+import com.iexec.core.replicate.ReplicatesService;
+import com.iexec.core.task.event.*;
+import com.iexec.core.task.update.TaskUpdateRequestConsumer;
+import com.iexec.core.task.update.TaskUpdateRequestManager;
+import com.iexec.core.worker.Worker;
+import com.iexec.core.worker.WorkerService;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.context.ApplicationEventPublisher;
+import org.springframework.stereotype.Service;
+
+import java.util.Date;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+import static com.iexec.core.task.TaskStatus.*;
+
+@Service
+@Slf4j
+public class TaskUpdateManager implements TaskUpdateRequestConsumer  {
+    private final TaskService taskService;
+    private final TaskRepository taskRepository;
+    private final TaskUpdateRequestManager taskUpdateRequestManager;
+    private final IexecHubService iexecHubService;
+    private final ReplicatesService replicatesService;
+    private final ApplicationEventPublisher applicationEventPublisher;
+    private final WorkerService workerService;
+    private final BlockchainAdapterService blockchainAdapterService;
+
+    public TaskUpdateManager(TaskService taskService,
+                             TaskRepository taskRepository,
+                             TaskUpdateRequestManager taskUpdateRequestManager,
+                             IexecHubService iexecHubService,
+                             ReplicatesService replicatesService,
+                             ApplicationEventPublisher applicationEventPublisher,
+                             WorkerService workerService,
+                             BlockchainAdapterService blockchainAdapterService) {
+        this.taskService = taskService;
+        this.taskRepository = taskRepository;
+        this.taskUpdateRequestManager = taskUpdateRequestManager;
+        this.iexecHubService = iexecHubService;
+        this.replicatesService = replicatesService;
+        this.applicationEventPublisher = applicationEventPublisher;
+        this.workerService = workerService;
+        this.blockchainAdapterService = blockchainAdapterService;
+        this.taskUpdateRequestManager.setRequestConsumer(this);
+    }
+
+    /**
+     * Async called when a task update request is received
+     */
+    @Override
+    public void onTaskUpdateRequest(String chainTaskId) {
+        log.info("Received task update request [chainTaskId:{}]", chainTaskId);
+        this.updateTaskRunnable(chainTaskId);
+    }
+
+    public CompletableFuture<Boolean> publishUpdateTaskRequest(String chainTaskId) {
+        return taskUpdateRequestManager.publishRequest(chainTaskId);
+    }
```

Comment:
Why do we use the same service to publish and consume at the same time? I think `publishUpdateTaskRequest()` is redundant, we should use `TaskUpdateRequestManager` directly to publish an update request.

Reply:
That was the previous behavior in `TaskService` that I only copied. See: https://github.com/iExecBlockchainComputing/iexec-core/blob/6375324c701cf36b858c7483d716ba34107bc2ee/src/main/java/com/iexec/core/task/TaskService.java#L209

I don't exactly know why it has been designed this way. @jeremyjams, any clue on that?
If there's no reason to that publish/consume behavior to happen there, we could certainly remove it.

Reply:
Ok, we can keep it that way in this PR, and eventually change it after.

Reply:
We can either 
(1) Merge TaskUpdateManager & TaskUpdateRequestManager (but testing is trickier)
or
(2) Use TaskUpdateRequestManager as a botleneck/filter (feature) between any client service and the TaskUpdateManager. Doing so, no one but the TaskUpdateRequestManager will be able to talk to the TaskUpdateManager  (to ensures TaskUpdateManager is fed properly)